packages/gatsby/src/utils/gatsby-webpack-eslint-config-reload.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * The problem: after GraphQL schema rebuilds, eslint loader keeps validating against
+ * the old schema.
+ *
+ * This plugin replaces options of eslint-plugin-graphql during develop
+ */
+const { store } = require(`../redux`)
+const eslintConfig = require(`./eslint-config`)
+const hasLocalEslint = require(`./local-eslint-config-finder`)
+
+const isEslintRule = rule => {
+  const options = rule && rule.use && rule.use[0] && rule.use[0].options
+  return options && typeof options.useEslintrc !== `undefined`
+}
+
+class GatsbyWebpackEslintConfigReload {
+  schema = null
+
+  constructor(options) {
+    this.plugin = { name: `GatsbyWebpackEslintConfigReload` }
+    this.options = options || {}
+  }
+
+  findEslintOptions = compiler =>
+    compiler.options.module.rules.find(isEslintRule).use[0].options
+
+  apply(compiler) {
+    compiler.hooks.compile.tap(this.plugin, stats => {
```

Comment:
This feels like it should be the string and not the object but I might be wrong.
```suggestion
    compiler.hooks.compile.tap(this.plugin.name, stats => {
```

Reply:
You are right, thanks for noticing!

Reply:
I used `GatsbyWebpackStatsExtractor` as a prototype for this plugin and it passes an object. But webpack docs say it should be a string. I'll change both plugins then.